### PR TITLE
V1 products

### PR DESCRIPTION
**Pull request by svenlombaert**
Documentation of the Products endpoints, V1

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -879,6 +879,193 @@ Delete a file
...
+
+### /addProduct.php [POST /addProduct.php]
+
+Add product
+
++ Request (application/x-www-form-urlencoded)
+
+    + Attributes (object)
+        + name: `Johan` (string, required) - The name of the product
+        + price: `12.34` (string, required) - The price of the new product
```
Comment (jenssegers) [inline apiary.apib]: Number?

File: apiary.apib [hunk 2]
```diff
@@ -879,6 +879,193 @@ Delete a file
             "status": "success"
         }
 
+## Products [/products]
+
+### /addProduct.php [POST /addProduct.php]
+
+Add product
+
++ Request (application/x-www-form-urlencoded)
```
Comment (jenssegers) [inline apiary.apib]: application/json

File: apiary.apib [hunk 3]
```diff
@@ -879,6 +879,193 @@ Delete a file
...
+                + 00
+                + 06
+                + 12
+                + 21
+                + CM
+                + EX
+                + MC
+                + VCMD
+        + account_id: `1234` (number, optional) - The ID of the bookkeeping account
+        + automerge_by_name: `true` (boolean, optional) - If this flag is set to 1, Teamleader will merge this info into an existing product with the same name, if it finds any.
```
Comment (jenssegers) [inline apiary.apib]:
```
automerge_by_name: `0` (enum[number])
            + Members
                + 0
                + 1
```

File: apiary.apib (hunk 2 above)
Comment (svenlombaert) [inline apiary.apib]: So, both the requests and the responses?

File: apiary.apib [hunk 4]
```diff
@@ -879,6 +879,193 @@ Delete a file
...
+            + Default: false
+        + purchase_price: `12.34` (number, optional) - The purchase price of the product
+        + custom_field_ID: `any` (string, optional) - Replace ID by the ID of your custom field. (eg custom_field_310)
+
+    + Body
```
Comment (jenssegers) [inline apiary.apib]: Can go if we use json I guess.

File: apiary.apib [hunk 5]
```diff
@@ -879,6 +879,193 @@ Delete a file
...
+
++ Request (application/x-www-form-urlencoded)
+
+    + Attributes (object)
+        + product_id: `1234` (string, required) - Id of the product
+        + name: `Johan` (string, optional) - The name of the product
+        + price: `12.34` (string, optional) - The price of the new product
+        + external_id: `1234A` (string, optional) - External ID code
+        + stock_amount: `100` (number, optional) - The new value for stock
+        + purchase_price: `12.34` (number, optional) - The purchase price of the product
```
Comment (jenssegers) [inline apiary.apib]: String here?

File: apiary.apib [hunk 6]
```diff
@@ -879,6 +879,193 @@ Delete a file
...
+
+
+### /deleteProduct.php [POST /deleteProduct.php]
+
+Delete product
+
++ Request (application/x-www-form-urlencoded)
+
+    + Attributes (object)
+        + product_id: `1234` (string, required) - Id of the product
```
Comment (jenssegers) [inline apiary.apib]: Number?

File: apiary.apib [hunk 7]
```diff
@@ -879,6 +879,195 @@ Delete a file
...
+    For example, first price returned will be `price_value_1` and the second will be `price_value_2`.
+
+    + Attributes (object)
+        + id: 203052 (number)
+        + external_id: ext_id_827 (string)
+        + name: Nice product (string)
+        + vat: 19 (string)
+        + booking_account: 700 (number)
+        + booking_account_id: 10 (number)
+        + price_value_1: 30.0000 (number)
```
Comment (jenssegers) [inline apiary.apib]: String here as well, or number?

File: apiary.apib [hunk 8]
```diff
@@ -879,6 +879,195 @@ Delete a file
...
+                + CM
+                + EX
+                + MC
+                + VCMD
+        + account_id: `1234` (number, optional) - The ID of the bookkeeping account
+        + automerge_by_name: `0` (enum[number]) - If this flag is set to 1, Teamleader will merge this info into an existing product with the same name, if it finds any.
+            + Members
+                + 0
+                + 1
+        + purchase_price: `12.34` (number, optional) - The purchase price of the product
```
Comment (jenssegers) [inline apiary.apib]: String here?

File: apiary.apib [hunk 9]
```diff
@@ -879,6 +879,195 @@ Delete a file
...
+### /getProduct.php [POST /getProduct.php]
+
+Get product details
+
++ Request (application/x-www-form-urlencoded)
+
+    + Attributes
+        + product_id: 20 (number, required) - The id of the product
+        + include_all_custom_fields: 1 (number, optional) - If this flag is set to 1, all custom fields will be fetched 
+        + include_photos: 1 (number, optional) - If this flag is set to 1, the product image url will be fetched
```
Comment (jenssegers) [inline apiary.apib]:
```
(enum[number])
            + Members
                + 0
                + 1
```

File: apiary.apib [hunk 10]
```diff
@@ -879,6 +879,195 @@ Delete a file
...
+
+            name=Johan&price=12.34&external_id=1234A&vat=00&account_id=1234&automerge_by_name=1&purchase_price=12.34&custom_field_310=test
+
+
+
++ Response 200 (application/json)
+    + Attributes (object)
+        + id: `416654` (number) - ID of the product that was added
+
+    + Body
```
Comment (jenssegers) [inline apiary.apib]: Can go if json.

File: apiary.apib [hunk 11]
```diff
@@ -879,6 +879,195 @@ Delete a file
...
+                + VCMD
+        + account_id: `1234` (number, optional) - The ID of the bookkeeping account
+        + automerge_by_name: `0` (enum[number]) - If this flag is set to 1, Teamleader will merge this info into an existing product with the same name, if it finds any.
+            + Members
+                + 0
+                + 1
+        + purchase_price: `12.34` (number, optional) - The purchase price of the product
+        + custom_field_ID: `any` (string, optional) - Replace ID by the ID of your custom field. (eg custom_field_310)
+
+    + Body
```
Comment (jenssegers) [inline apiary.apib]: Can go if json.

File: apiary.apib [hunk 12]
```diff
@@ -879,6 +879,195 @@ Delete a file
...
+    + Attributes (object)
+        + product_id: `1234` (string, required) - Id of the product
+        + name: `Johan` (string, optional) - The name of the product
+        + price: `12.34` (string, optional) - The price of the new product
+        + external_id: `1234A` (string, optional) - External ID code
+        + stock_amount: `100` (number, optional) - The new value for stock
+        + purchase_price: `12.34` (number, optional) - The purchase price of the product
+        + custom_field_ID: `any` (string, optional) - Replace ID by the ID of your custom field. (eg custom_field_310)
+
+    + Body
```
Comment (jenssegers) [inline apiary.apib]: Can go if json.

File: apiary.apib [hunk 13]
```diff
@@ -879,6 +879,195 @@ Delete a file
...
+### /deleteProduct.php [POST /deleteProduct.php]
+
+Delete product
+
++ Request (application/json)
+
+    + Attributes (object)
+        + product_id: `1234` (string, required) - Id of the product
+
+    + Body
```
Comment (jenssegers) [inline apiary.apib]: Can go if json.

File: apiary.apib [hunk 14]
```diff
@@ -879,6 +879,121 @@ Delete a file
...
+## Products [/products]
+
+### /addProduct.php [POST /addProduct.php]
+
+Add product
+
++ Request (application/json)
+
+    + Attributes (object)
+        + name: `Johan` (string, required) - The name of the product
```
Comment (steftrenson) [inline apiary.apib]: PP: Johan is not a nice example of a product name

File: apiary.apib [hunk 15]
```diff
@@ -879,6 +879,121 @@ Delete a file
...
+
+### /updateProduct.php [POST /updateProduct.php]
+
+Update product
+
++ Request (application/json)
+
+    + Attributes (object)
+        + product_id: `1234` (number, required) - Id of the product
+        + name: `Johan` (string, optional) - The name of the product
```
Comment (steftrenson) [inline apiary.apib]: PP: same thing here

File: apiary.apib [hunk 16]
```diff
@@ -879,6 +879,121 @@ Delete a file
...
+        + status: `success` (string) - Indicating success or failure of the update operation.
+
+### /getProducts.php [POST /getProducts.php]
+
+Search all products
+
++ Request (application/json)
+
+    + Attributes
+        + amount: 10 (number, required) - The amount of products returned per request
```
Comment (steftrenson) [inline apiary.apib]: 10 should be `10` ? Same for following attributes

File: apiary.apib [hunk 17]
```diff
@@ -879,6 +879,121 @@ Delete a file
...
+        + modifiedsince: 1498056035 (number, optional) - Unix timestamp. Teamleader will only return products that have been added or modified since that timestamp
+
++ Response 200 (application/json)
+    
+    Prices will be returned as `price_value_<index_of_price>`, they are suffixed with the price index. 
+    For example, first price returned will be `price_value_1` and the second will be `price_value_2`.
+
+    + Attributes (array)
+        + (object)
+            + id: 203052 (number)
```
Comment (Steerlin) [inline apiary.apib]: back-ticks ?

Comment (Steerlin) [inline apiary.apib]: nevermind - is response

File: apiary.apib [hunk 18]
```diff
@@ -1007,3 +1122,28 @@ Delete a file
 + date_lost_formatted: 16/02/2017 (string)
 + reason_refused: `No interest` (string)
 + source_id: `8799` (number)
+
+## Product (object)
```
Comment (jenssegers) [inline apiary.apib]: PP: Rename to ProductDetail to follow the other data structures

**Pull request merged**